plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -67,9 +70,32 @@ data DefaultFun
     | EncodeUtf8
     | DecodeUtf8
     | Trace
-    | Nop1  -- TODO. These are only used for costing calibration and shouldn't be included in the defaults.
+    -- TODO. These are only used for costing calibration and shouldn't be included in the defaults.
+    | Nop1
     | Nop2
     | Nop3
+    | Fst
+    | Snd
+    | Null
+    | Head
+    | Tail
+    -- Functions somehow corresponding to constructors of Haskell data types has the @C@ suffix,
+    -- unless there's an established name for the projection function or the constructor does not
+    -- have an ASCII name (both conditions apply to 'Head' and 'Tail', for example).
```

Comment:
> I'd still rather include the name of the type they deconstruct in here somewhere, otherwise it's very non-obvious!

But they don't have the type included in the actual constructors of the data type, so why should we include them here? At the call site `Constr` and `ConstrC` are equally "type-unspecified".

Reply:
Because you usually use `Constr` in the context of a Haskell expression where the typechecker will correct you if you get it wrong. Also it's probably defined in a separate module, and can be used qualified if you like. 

As a human being reading this list of builtin function identifiers, it's just not obvious what this floating name is supposed to be. I guess to figure it out you have to go look at the default implementation.

Reply:
Ok, I don't really like it, but I've added suffixes to all list-, pair- and data-related builtins.